Firestore/core/src/model/value_util.cc

```diff
@@ -600,35 +605,38 @@ bool Contains(google_firestore_v1_ArrayValue haystack,
   return false;
 }
 
-Message<google_firestore_v1_Value> NullValue() {
-  Message<google_firestore_v1_Value> null_value;
-  null_value->which_value_type = google_firestore_v1_Value_null_value_tag;
-  null_value->null_value = {};
+google_firestore_v1_Value NullValue() {
+  google_firestore_v1_Value null_value;
+  null_value.which_value_type = google_firestore_v1_Value_null_value_tag;
+  null_value.null_value = {};
   return null_value;
 }
 
 bool IsNullValue(const google_firestore_v1_Value& value) {
   return value.which_value_type == google_firestore_v1_Value_null_value_tag;
 }
 
-nanopb::Message<google_firestore_v1_Value> MinValue() {
-  return NullValue();
+google_firestore_v1_Value MinValue() {
+  google_firestore_v1_Value null_value;
+  null_value.which_value_type = google_firestore_v1_Value_null_value_tag;
+  null_value.null_value = {};
+  return null_value;
 }
 
 bool IsMinValue(const google_firestore_v1_Value& value) {
   return IsNullValue(value);
 }
 
-Message<google_firestore_v1_Value> MaxValue() {
-  Message<google_firestore_v1_Value> max_value;
-  max_value->which_value_type = google_firestore_v1_Value_map_value_tag;
-  max_value->map_value.fields_count = 1;
-  max_value->map_value.fields =
+google_firestore_v1_Value MaxValue() {
+  google_firestore_v1_Value max_value;
+  max_value.which_value_type = google_firestore_v1_Value_map_value_tag;
+  max_value.map_value.fields_count = 1;
+  max_value.map_value.fields =
       nanopb::MakeArray<google_firestore_v1_MapValue_FieldsEntry>(1);
-  max_value->map_value.fields[0].key = nanopb::MakeBytesArray("__type__");
-  max_value->map_value.fields[0].value.which_value_type =
+  max_value.map_value.fields[0].key = nanopb::MakeBytesArray("__type__");
```

Comment:
Hm, this is also leaking memory now. This should also use a static instance.

Reply:
Fixed.